src/Streamly/Internal/Prelude.hs

```diff
@@ -751,10 +751,9 @@ repeatMSerial = fromStreamS . S.repeatM
 -- @
 --
 -- @since 0.1.2
-iterate :: IsStream t => (a -> a) -> a -> t m a
-iterate step = K.fromStream . go
-    where
-    go s = K.cons s (go (step s))
+{-# INLINE_EARLY iterate #-}
```

Comment:
A simple "INLINE" should be enough here, we do not need INLINE_EARLY as we do not have a rewrite rule here.